pkg/reconciler/managed/reconciler.go

```diff
@@ -880,7 +880,7 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (resu
 		log.Debug("Reconciliation is paused either through the `spec.managementPolicies` or the pause annotation", "annotation", meta.AnnotationKeyReconciliationPaused)
 		record.Event(managed, event.Normal(reasonReconciliationPaused, "Reconciliation is paused either through the `spec.managementPolicies` or the pause annotation",
 			"annotation", meta.AnnotationKeyReconciliationPaused))
-		managed.SetConditions(xpv1.ReconcilePaused())
+		managed.SetConditions(xpv1.ReconcilePaused().WithObservedGeneration(managed.GetGeneration()))
```

Comment:
Is there ever a case where we want to set a condition inside a reconciler, but _not_ its generation to the generation of the resource?

```go
cm := resource.NewConditionManager(managed)
cm.SetConditions(...)
```

I'm imagining something like this that'd wrap `SetConditions` and automatically propagate the generation from `managed` to the conditions being set. I could be convinced though that this isn't worth doing now if the plan is to move to a more heavyweight condition manager.

Reply:
I had that same thought and we can get there, but I was not sure yet about how all the interfaces work inside of crossplane so I did it hard mode and added the tests to make sure it is working and then I can look at removing stuff as we tighten up condition accessors.

Reply:
> move to a more heavyweight condition manager

This is my plan, yes. But I need more context on the project before I can understand what the project needs :D

Reply:
Done. But we should land https://github.com/crossplane/crossplane-runtime/pull/831/f first and then I can rebase.